src/Plug.vala

```diff
@@ -367,10 +370,45 @@ namespace Power {
                 grid.attach (dim_label, 0, 0, 1, 1);
                 grid.attach (dim_switch, 1, 0, 1, 1);
             }
-
             return grid;
         }
 
+        private static bool lid_detect () {
+            var lid_path = File.new_for_path ("/proc/acpi/button/lid/");
+            var enumerator = lid_path.enumerate_children (
+                GLib.FileAttribute.STANDARD_NAME,
+                FileQueryInfoFlags.NONE);
+            try {
+                FileInfo lid;
```

Comment:
You don't need to allocate memory for this variable here. You could just do: `if (enumerator.next_file () != null) {`